test/sql.test.js

```diff
@@ -128,6 +128,18 @@ describe('sql connector', function() {
     });
   });
 
+  it('builds where with regexp', function() {
+    var where = connector.buildWhere('customer', {
+      name: {
+        regexp: '^J'
+      }
+    });
+    expect(where.toJSON()).to.eql({
+      sql: 'WHERE `NAME` REGEXP ?',
+      params: ['^J']
+    });
+  });
+
```

Comment:
We should add one more test to cover the case where regexp takes an instance of RegExp, such as `/^J/i`.